File: tools/parity/dispatch/git-cquoted-path.ts
// TS dispatch for the git-cquoted-path parity module: maps the shared vector
// function names to the real `src/shared/git-cquoted-path.ts` export so the
// harness compares the live TS reference against the Rust port.

import { decodeGitCQuotedPath } from '../../../src/shared/git-cquoted-path'

export function dispatch(fn: string, input: unknown): unknown {
  switch (fn) {
    case 'decodeGitCQuotedPath':
      return decodeGitCQuotedPath(input as string)
    default:
      throw new Error(`unknown function ${fn}`)
  }
}

File: rust/crates/orca-parity/src/modules/git_cquoted_path.rs
//! Parity dispatch for `orca_core::git_cquoted_path` vs
//! `src/shared/git-cquoted-path.ts`.

use orca_core::git_cquoted_path::decode_git_cquoted_path;
use serde_json::{json, Value};

pub fn dispatch(function: &str, input: &Value) -> Value {
    match function {
        "decodeGitCQuotedPath" => match input.as_str() {
            Some(value) => Value::String(decode_git_cquoted_path(value)),
            // Vectors only carry string inputs; a non-string is a vector bug.
            None => json!({ "__parity_error__": "expected string input" }),
        },
        other => json!({ "__parity_error__": format!("unknown function {other}") }),
    }
}
